packages/flutter_tools/lib/src/commands/attach.dart

```diff
@@ -215,7 +215,11 @@ known, it can be explicitly provided to attach via the command-line, e.g.
   Future<FlutterCommandResult> runCommand() async {
     await _validateArguments();
 
-    final Device device = await (findTargetDevice() as FutureOr<Device>);
```

Comment:
all these `FutureOr` casts were deemed unneccessary by the compiler after opting in to null safety